rust/src/coordinator.rs

```diff
@@ -317,32 +317,60 @@ impl Coordinator {
         self.phase = Phase::Sum;
     }
 
-    /// End the sum phase and proceed to the update phase. Fails due to insufficient sum
-    /// participants.
-    fn proceed_phase_update(&mut self) -> Result<(), PetError> {
-        self.freeze_dict_sum()?;
+    /// Check whether there are enough sum participants for starting
+    /// the update phase.
+    fn has_enough_sum_participants(&self) -> bool {
```

Comment:
i would suggest a more consistent naming for those three methods, like
```
has_enough_sums()
has_enough_updates()
has_enough_masks()
```
or alternatively
```
has_enough_sum_messages()
has_enough_update_messages()
has_enough_sum2_messages()
```

Reply:
ah very nice. I have a slight preference the first version. How about you (or @finiteprods) ?

Reply:
same here

Reply:
same

Reply:
alright it's settled then.